decision/blockstoremanager.go

```diff
@@ -0,0 +1,114 @@
+package decision
+
+import (
+	"context"
+	"sync"
+
+	blocks "github.com/ipfs/go-block-format"
+	cid "github.com/ipfs/go-cid"
+	bstore "github.com/ipfs/go-ipfs-blockstore"
+	process "github.com/jbenet/goprocess"
+)
+
+// blockstoreManager maintains a pool of workers that make requests to the blockstore.
+type blockstoreManager struct {
+	bs          bstore.Blockstore
+	workerCount int
+	jobs        chan func()
+}
+
+// newBlockstoreManager creates a new blockstoreManager with the given context
+// and number of workers
+func newBlockstoreManager(ctx context.Context, bs bstore.Blockstore, workerCount int) *blockstoreManager {
+	return &blockstoreManager{
+		bs:          bs,
+		workerCount: workerCount,
+		jobs:        make(chan func(), 1),
+	}
+}
+
+func (bsm *blockstoreManager) start(ctx context.Context, px process.Process) {
```

Comment:
* We don't really need the ids, do we?
* Any reason to not just pass the process into the worker directly? We can skip the context.